pkg/apis/security/v1alpha1/common_types.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2020 Google LLC.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+)
+
+// StringMatch defines the specification to match a string.
+type StringMatch struct {
+	// Exact is to match the exact string.
+	Exact string `json:"exact,omitempty"`
+
+	// Prefix is to match the prefix of the string.
+	Prefix string `json:"prefix,omitempty"`
+
+	// Suffix is to match the suffix of the string.
+	Suffix string `json:"suffix,omitempty"`
+
+	// Presence is to match anything but empty.
+	Presence bool `json:"presence,omitempty"`
```

Comment:
I'm assuming that you have Presence here separately from Prefix because you don't want to express Presence as `Prefix: ""` (which would probably be serialized as the empty string).